Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -527,21 +550,98 @@ extension DefaultPublisher {
         }
     }
     
-    private func saveLocationForFurtherSending(trackableId: String, location: EnhancedLocationUpdate) {
-        trackableState.skippedLocationsAdd(for: trackableId, location: location)
+    private func sendRawLocationUpdate(event: RawLocationChangedEvent, trackable: Trackable) {        
+        lastRawLocations[trackable] = event.locationUpdate.location
+        lastRawTimestamps[trackable] = event.locationUpdate.location.timestamp
+
+        event.locationUpdate.skippedLocations = rawLocationState.locationsList(for: trackable.id).map { $0.location }
+
+        rawLocationState.markMessageAsPending(for: trackable.id)
+
+        ablyPublisher.sendRawLocation(location: event.locationUpdate, trackable: trackable) { [weak self] result in
+            switch result {
+            case .failure(let error):
+                self?.enqueue(event: SendRawLocationFailureEvent(error: error, locationUpdate: event.locationUpdate, trackable: trackable))
+            case .success:
+                self?.enqueue(event: SendRawLocationSuccessEvent(trackable: trackable, location: event.locationUpdate.location))
+            }
+        }
+    }
+    
+    private func saveEnhancedLocationForFurtherSending(trackableId: String, location: EnhancedLocationUpdate) {
+        enhancedLocationState.addLocation(for: trackableId, location: location)
+    }
+    
+    private func saveRawLocationForFurtherSending(trackableId: String, location: RawLocationUpdate) {
+        rawLocationState.addLocation(for: trackableId, location: location)
+    }
+    
+    private func performRawLocationChanged(_ event: RawLocationChangedEvent) {
+        guard !state.isStoppingOrStopped else {
+            logger.error("Cannot perform EnhancedLocationChangedEvent. Publisher is not working.")
+            return
+        }
+
+        let trackablesToSend = trackables.filter { trackable in
+            guard !rawLocationState.hasPendingMessage(for: trackable.id) else {
+                rawLocationState.addToWaiting(locationUpdate: event.locationUpdate, for: trackable.id)
+                return false
+            }
+            
+            let shouldSend = shouldSendLocation(
+                location: event.locationUpdate.location,
+                lastLocation: lastRawLocations[trackable],
+                lastTimestamp: lastRawTimestamps[trackable],
+                resolution: resolutions[trackable]
+            )
+            
+            if !shouldSend {
+                rawLocationState.addLocation(for: trackable.id, location: event.locationUpdate)
+            }
+            
+            return shouldSend
+        }
+
+        if areRawLocationsEnabled == true {
+            trackablesToSend.forEach { trackable in
+                sendRawLocationUpdate(event: event, trackable: trackable)
+            }
+        }
```

Comment:
Here you are performing some logic to filter the trackables and check if a raw location update for them should be sent even if the raw locations are disabled :wink: I'd say that the "are raw locations enabled" check should be done as soon as possible, so we won't waste performance on things we don't need :thinking:

Reply:
5d8105d928c5c22aacee7c83b57e3c98089c098e

Reply:
It's better but now you would report that the publisher has failed to send a message because it is not working, while in the reality it can't send it because it's simply disabled :wink: So I'd move it to a separate guard with its own error log message.

Reply:
aa699882c9338a466ad8ab6a1f8555a15b5796b6